### ISSUE TITLE
Running on iOS simulator

### ISSUE THREAD

**Issue by awelc** [frida/frida#1830]
It seems like Frida can be used with the iOS simulators but I could not find a comprehensive guide on how to make that happen, particularly when using newer versions of MacOS, that seem to come with more and more restrictions for developers - in my case it's BigSur 11.5.2. I am also using Xcode 12.5.1 and trying this with simulated iPhone 12 (iOS 14.5).

I was trying to follow unofficial instructions that I found [here](https://gist.github.com/rustymagnet3000/605c333519cd265c7eac9d556f46dc75), that is:

```
// demoapp is the iOS app name
myapp=$(ps x | grep -i -m1 demoapp | awk '{print $1}')
frida-trace -i "getfsent*" -p $myapp
```

These instructions make me believe that using Frida for iOS apps on a simulator is more like using it for MacOS apps, but trying to attach to a very simple app created in Xcode (debug configuration) always results in:

`Failed to attach: process with pid XXXXXX either refused to load frida-agent, or terminated during injection`

This is similar to another [issue](https://github.com/frida/frida/issues/1829) created recently, but I am not trying to work with arbitrary apps, but rather specifically iOS apps on a simulator, and I hoped that either what I am seeing is a known problem (possibly with an ongoing work on a solution) or perhaps there is some known workaround.

While I can indeed see the same message for some other MacOS apps (particularly built-in ones) I can without too much trouble (need root...) attach and trace apps installed via Brew (e.g., the Emacs app).

If anyone can suggest how to overcome this for iOS simulated apps, I would greatly appreciate it.

**Comment by awelc** [comment]
A bit of an update on this (and my apologies - I am pretty new to this...). When attempting to attach to the process, this is the error I am seeing:

```
Frida: agent.vala:140: Unable to start agent: Unable to fetch file descriptor from re.frida.piped.23068 (bootstrap_look_up returned 'unknown error code')
```

**Comment by oleavr** [comment]
This should now be fixed in Frida 15.1.0.
